builders/get-started/eth-compare/security.md

```diff
@@ -0,0 +1,108 @@
+
+## Introduction {: #introduction }
+
+When developing smart contracts on Moonbeam, there are some security considerations to be aware of that do not apply when developing on Ethereum. Moonbeam has several [precompiled contracts](/builders/pallets-precompiles/precompiles/){target=_blank}, which are Solidity interfaces that enable developers to access Substrate-based functionality through the Ethereum API, but circumventing the EVM. Although the precompiled contracts are designed to improve the developer experience, there can be some unintended consequences that must be considered.
+
+This guide will outline and provide examples of some security considerations to be cognizant of when developing on Moonbeam.
+
+## Arbitrary Code Execution {: #arbitrary-code-execution }
+
+Arbitrary code execution in Solidity is the ability to execute code and call functions of other contracts using an arbitrary number of arguments of any type.
+
+A smart contract allows arbitrary execution of another contract when it allows a user to invoke the `call()` function and pass in arbitrary call data. The [`call()` function](https://solidity-by-example.org/call/){target=_blank} is made available through the [address data type in Solidity](https://docs.soliditylang.org/en/latest/types.html#address){target=_blank}. When the `call()` function is invoked, the target contract is called using the arbitrary call data.
+
+An example of arbitrary code execution, in it's simplest form, can be seen in the following diagram.
+
+![Arbitrary code execution](/images/builders/get-started/eth-compare/security/security-1.png)
+
+As previously mentioned, one major concern of arbitrarily executing code on Moonbeam is that Moonbeam has precompile contracts that can be called, which can be used to get around some protections that are typically available on Ethereum. To safely use arbitrary code execution on Moonbeam, you should consider the following, which **only applies to contracts that allow arbitrary code execution**:
+
+- Moonbeam [precompiled contracts](builders/pallets-precompiles/precompiles/){target=_blank} such as the Native ERC-20 precompile, XC-20 precompiles, and XCM-related precompiles allow users to manage and transfer assets without requiring access to the EVM. Instead, these actions are done using the Substrate API. So, if your contract holds native tokens or XC-20s and allows arbitrary code execution, these precompiles can be used to drain the balance of the contract, bypassing any security checks that are normally enforced by the EVM
+- Setting the value attribute of the transaction object to a fixed amount when using the `call()` function (for example, `call{value: 0}(...)`) can be bypassed by calling the native asset precompile and specifying an amount to transfer in the encoded call data
+- Providing the function selector for the function that you want to allow to be executed is considered safe, assuming that the function call is safe and does not call precompiles
```

Comment:
Is this saying that the contract should provide the fn selector or that it's safe in some cases for the user to provide it? If the former, maybe say "hard coding the selector"?

Or instead of saying what's safe, say what isn't -- which I think is the combination of user-influence-able target (or any target that is our custom precompiles) and a fn selector.